### PR TITLE
Regional model recipe improvements

### PR DESCRIPTION
**Pull request by jleinonen**
# PhysicsNeMo Pull Request

## Description

Adds a few usability and performance improvements to the regional model recipe:

1. Allow passing `attn_kwargs` to `DiT` models from config files in `model.hyperparameters.attn_kwargs`.
2. Increase TensorBoard logging buffer size and rely on TensorBoard's internal async log writing rather than dumping logs manually. This consolidates log writes to fewer, bigger writes, reducing congestion on networked file systems.
3. Set `pin_memory=False` for validation dataloader; this reduces CPU memory utilization without a performance hit.
4. Fix WandB logging issue where previously logged images were re-logged.

## Checklist

- [x] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA/physicsnemo/blob/main/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.
- [ ] The [CHANGELOG.md](https://github.com/NVIDIA/physicsnemo/blob/main/CHANGELOG.md) is up to date with these changes.
- [ ] An [issue](https://github.com/NVIDIA/physicsnemo/issues) is linked to this pull request.
- [ ] If I am implementing a new model or modifying any existing model, I have followed the [Models Implementation Coding Standards](https://github.com/NVIDIA/physicsnemo/wiki/Coding-standards-for-models-implementation).

## Review Process

**All PRs are reviewed by the PhysicsNeMo team before merging.**

Depending on which files are changed, GitHub may automatically assign a maintainer for review.

We are also testing AI-based code review tools (e.g., Greptile), which may add automated comments with a confidence score.
This score reflects the AI’s assessment of merge readiness and is **not** a qualitative judgment of your work, nor is
it an indication that the PR will be accepted / rejected.

AI-generated feedback should be reviewed critically for usefulness.
You are not required to respond to every AI comment, but they are intended to help both authors and reviewers.
Please react to Greptile comments with 👍 or 👎 to provide feedback on their accuracy.

### CODE REVIEW

File: examples/weather/regional_weather_diffusion/utils/logging.py [hunk 1]
```diff
@@ -71,7 +71,7 @@ def __init__(self, name: str, cfg: MainConfig):
         if self.log_to_tensorboard:
             tb_dir = os.path.join(cfg.training.rundir, "tensorboard")
             os.makedirs(tb_dir, exist_ok=True)
-            self.tensorboard_writer = SummaryWriter(log_dir=tb_dir)
+            self.tensorboard_writer = SummaryWriter(log_dir=tb_dir, max_queue=65536)
```
[the remark this answers is not in the document]
Comment (jleinonen) [reply examples/weather/regional_weather_diffusion/utils/logging.py:74]: Validation usually runs rarely and during training only scalars are logged. So there should only be a few images in the buffer and mostly scalars. Therefore 65535 logged items shouldn't take up lots of memory.

**Review by pzharrington — APPROVED** [review]
LGTM
